api/execute_and_listen.go

```diff
@@ -0,0 +1,24 @@
+package api
+
+import (
+	"github.com/mesg-foundation/core/execution"
+	uuid "github.com/satori/go.uuid"
+)
+
+// ExecuteAndListen executes given task and listenes for result.
+func (a *API) ExecuteAndListen(serviceID, task string, inputs map[string]interface{}) (*execution.Execution, error) {
+	tag := uuid.NewV4().String()
+	_, err := a.ExecuteTask(serviceID, task, inputs, []string{tag})
+	if err != nil {
+		return nil, err
+
+	}
+	result, err := a.ListenResult(serviceID, ListenResultTagFilters([]string{tag}))
```

Comment:
I think the `ListenResult` needs to be executed before the `ExecuteTask`.

In the command execute there is also a timeout between Listen and Execute to be sure that the Listen stream is really up and running.

https://github.com/mesg-foundation/core/blob/ff624f46db6e13bab950a96a11d025c182c1e74a/commands/service_execute.go#L78-L88

Reply:
done